File: app/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/InputBgTest.java
package info.nightscout.androidaps.plugins.general.automation.elements;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import info.AAPSMocker;
import info.nightscout.androidaps.Constants;
import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.plugins.configBuilder.ProfileFunctions;

@RunWith(PowerMockRunner.class)
@PrepareForTest({MainApp.class, ProfileFunctions.class})
public class InputBgTest {

    @Test
    public void getSetValueTest() {
        InputBg i = new InputBg().setUnits(Constants.MMOL).setValue(5d);
        Assert.assertEquals(5d, i.getValue(), 0.01d);
        Assert.assertEquals(InputBg.MMOL_MIN, i.minValue, 0.01d);
        i = new InputBg().setValue(100d).setUnits(Constants.MGDL);
        Assert.assertEquals(100d, i.getValue(), 0.01d);
        Assert.assertEquals(InputBg.MGDL_MIN, i.minValue, 0.01d);
        Assert.assertEquals(Constants.MGDL, i.getUnits());
    }


    @Before
    public void prepare() {
        AAPSMocker.mockMainApp();
        AAPSMocker.mockStrings();
        AAPSMocker.mockProfileFunctions();
    }
}

File: automation/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/InputBgTest.kt
package info.nightscout.androidaps.plugins.general.automation.elements

import info.nightscout.androidaps.Constants
import info.nightscout.androidaps.interfaces.GlucoseUnit
import info.nightscout.androidaps.plugins.general.automation.triggers.TriggerTestBase
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.`when`
import org.powermock.modules.junit4.PowerMockRunner

@RunWith(PowerMockRunner::class)
class InputBgTest : TriggerTestBase() {

    @Test
    fun setValueTest() {
        var i: InputBg = InputBg(profileFunction).setUnits(GlucoseUnit.MMOL).setValue(5.0)
        Assert.assertEquals(5.0, i.value, 0.01)
        Assert.assertEquals(InputBg.MMOL_MIN, i.minValue, 0.01)
        i = InputBg(profileFunction).setValue(100.0).setUnits(GlucoseUnit.MGDL)
        Assert.assertEquals(100.0, i.value, 0.01)
        Assert.assertEquals(InputBg.MGDL_MIN, i.minValue, 0.01)
        Assert.assertEquals(GlucoseUnit.MGDL, i.units)
    }

    @Before
    fun prepare() {
        `when`(profileFunction.getUnits()).thenReturn(GlucoseUnit.MGDL)
    }
}